tools/linter/adapters/header_only_linter.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+"""
+Checks that all symbols in torch/header_only_apis.txt are tested in a .cpp
+test file to ensure header-only-ness. The .cpp test file must be built
+without linking libtorch.
+"""
+
+import argparse
+import json
+from enum import Enum
+from pathlib import Path
+from typing import NamedTuple, Union
+
+
+LINTER_CODE = "HEADER_ONLY_LINTER"
+
+
+class LintSeverity(str, Enum):
+    ERROR = "error"
+    WARNING = "warning"
+    ADVICE = "advice"
+    DISABLED = "disabled"
+
+
+class LintMessage(NamedTuple):
+    path: Union[str, None]
+    line: Union[int, None]
+    char: Union[int, None]
```

Comment:
nit: why not Optional?

Reply:
This is following the pattern set in the other lint adapters (but they somehow manage to use | notation without mypy yelling at them)